clients/admin-ui/src/features/datamap/reporting/DatamapReportTable.tsx

```diff
@@ -273,6 +290,55 @@ export const DatamapReportTable = () => {
     setTotalPages(totalPages);
   }, [totalPages, setTotalPages]);
 
+  // Get custom fields
+  useGetAllCustomFieldDefinitionsQuery();
+  const customFields = useAppSelector(selectAllCustomFieldDefinitions);
+
+  const customFieldColumns = useMemo(() => {
+    // Determine custom field keys by
+    // 1. If they aren't in our expected, static, columns
+    // 2. If they start with one of the custom field prefixes
+    const datamapKeys = datamapReport
+      ? Object.keys(datamapReport.items[0])
+      : [];
+    const defaultKeys = Object.values(COLUMN_IDS);
+    const customFieldKeys = datamapKeys
+      .filter((k) => !defaultKeys.includes(k as COLUMN_IDS))
+      .filter(
+        (k) =>
+          k.startsWith(CUSTOM_FIELD_DATA_USE_PREFIX) ||
+          k.startsWith(CUSTOM_FIELD_SYSTEM_PREFIX)
+      );
+
+    // Create column objects for each custom field key
+    const columns = customFieldKeys.map((key) => {
+      // We need to figure out the original custom field object in order to see
+      // if the value is a string[], which would want `showHeaderMenu=true`
+      const customField = customFields.find((cf) =>
+        key.includes(_.snakeCase(cf.name))
+      );
+      const displayText = _.upperFirst(key.replaceAll("_", " "));
```

Comment:
this results in headers like 
1. `system_pokemon_party` ➡️ `System pokemon party`
2. `privacy_declaration_color` ➡️ `Privacy declaration color`

I'm not sure that we still call these "privacy declarations" in the UI anymore, so maybe we should convert this to i.e. "Data use color" here instead?